python/pyspark/sql/pandas/group_ops.py

```diff
@@ -366,6 +366,10 @@ class PandasCogroupedOps:
     """
 
     def __init__(self, gd1: "GroupedData", gd2: "GroupedData"):
+        gel1 = gd1._jgd.groupingExprs().length()
+        gel2 = gd2._jgd.groupingExprs().length()
+        assert gel1 == gel2, f"group keys must have same size: {gel1} != {gel2}"
```

Comment:
Nit: I thinking we can mention `Co-grouping` in the error message to better clarify this is from the cogroup operation?

Reply:
Good point, will add.

Reply:
Changed to `"Cogroup keys must have ..."`